File: 287-find-the-duplicate-number/find-the-duplicate-number.java
class Solution {
    public int findDuplicate(int[] nums) {
        Arrays.sort(nums);
        for(int i = 0; i < nums.length-1; i++)
        {
            
            if(Math.abs(nums[i]) - Math.abs(nums[i+1]) == 0)
            {
                return nums[i];
            }
        }
        return 0;
    }
}

File: 287-find-the-duplicate-number/find-the-duplicate-number.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var findDuplicate = function(nums) {
    let output = {}
    let outputNum = nums[0];
    let i = 0;
    while(i<nums.length) {
        output[nums[i]] = (output[nums[i]] || 0) + 1;
        console.log(output[nums[i]])
        if(output[nums[i]] > 1) {
            outputNum = nums[i];
            break;
        }
        i++;
    }
    return outputNum;
};